Source/WebKit/UIProcess/API/Cocoa/WKWebView.mm

```diff
@@ -455,31 +455,16 @@ - (void)_updateScreenTimeViewGeometry
     [[_screenTimeWebpageController view] setFrame:bounds];
 }
 
-- (void)_updateScreenTimeShieldVisibilityForWindow
+- (void)_updateScreenTimeBasedOnWindowVisibility
 {
     BOOL viewIsInWindow = !!self.window;
+    BOOL viewIsVisible = viewIsInWindow;
 
-    BOOL showsSystemScreenTimeBlockingView = [_configuration _showsSystemScreenTimeBlockingView];
-
-    if (viewIsInWindow) {
-        if (!showsSystemScreenTimeBlockingView && _screenTimeBlurredSnapshot) {
-            [_screenTimeBlurredSnapshot setHidden:NO];
-            RELEASE_LOG(ScreenTime, "Screen Time has updated visibility to show blurred view.");
-        } else if (showsSystemScreenTimeBlockingView) {
-            [[_screenTimeWebpageController view] setHidden:NO];
-            RELEASE_LOG(ScreenTime, "Screen Time has updated visibility to show system shield.");
-        }
-    } else {
-        if (_screenTimeBlurredSnapshot) {
-            [_screenTimeBlurredSnapshot setHidden:YES];
-            RELEASE_LOG(ScreenTime, "Screen Time has updated visibility to hide blurred view.");
-        } else if (showsSystemScreenTimeBlockingView) {
-            [[_screenTimeWebpageController view] setHidden:YES];
-            RELEASE_LOG(ScreenTime, "Screen Time has updated visibility to hide system shield.");
-        }
-    }
+    if (viewIsInWindow)
+        [self _installScreenTimeWebpageControllerIfNeeded];
+    else
+        [self _uninstallScreenTimeWebpageController];
 
-    BOOL viewIsVisible = viewIsInWindow;
 #if PLATFORM(MAC)
     viewIsVisible &= ((self.window.occlusionState & NSWindowOcclusionStateVisible) == NSWindowOcclusionStateVisible);
 #endif
```

Comment:
I think this logic should continue to be kept next to `viewIsVisible` (you can move it up).